.gemini/styleguide.md

```diff
@@ -0,0 +1,293 @@
+# qsim style guide
+
+This style guide outlines the coding conventions for this project. There are
+separate subsections for Python, C++, and other file types below.
+
+## General guidance
+
+### Overall principles
+
+*   _Readability_: Code should be easy to understand for all team members.
+
+*   _Maintainability_: Code should be easy to modify and extend.
+
+*   _Consistency_: Adhering to a consistent style across all projects improves
+    collaboration and reduces errors.
+
+*   _Performance_: While readability is paramount, code should be efficient.
+
+### Overall development approach
+
+*   Use test-driven development (TDD) whenever possible. Write a failing test
```

Comment:
I think test-driven development was never used for this project.

Reply:
Got it. The next push will remove the text about TDD.